data/crac/crac-impl/src/main/java/com/farao_community/farao/data/crac_impl/SimpleCnec.java

```diff
@@ -53,31 +55,33 @@ public NetworkElement getNetworkElement() {
         return networkElement;
     }
 
-    public double computeMargin(Network network) {
+    public double computeMargin(Network network, Unit requestedUnit) {
         // todo : switch units if no I is available but P is available
-        // todo : add a requested unit
         double flow;
-        Unit unit = thresholds.getUnit();
-        if (unit.equals(Unit.AMPERE)) {
+        if (requestedUnit.equals(Unit.AMPERE)) {
             flow = getI(network);
         } else {
             flow = getP(network);
         }
-        return Math.min(thresholds.getMaxThreshold(unit).orElse(Double.POSITIVE_INFINITY) - flow, flow - thresholds.getMinThreshold(unit).orElse(Double.NEGATIVE_INFINITY));
+        return Math.min(getMaxThreshold(requestedUnit).orElse(Double.POSITIVE_INFINITY) - flow, flow - getMinThreshold(requestedUnit).orElse(Double.NEGATIVE_INFINITY));
     }
 
     public void setNetworkElement(NetworkElement networkElement) {
         this.networkElement = networkElement;
     }
 
-    public AbstractThreshold getThresholds() {
+    public Set<AbstractThreshold> getThresholds() {
         return thresholds;
     }
 
-    public void setThresholds(AbstractThreshold thresholds) {
+    public void setThresholds(Set<AbstractThreshold> thresholds) {
         this.thresholds = thresholds;
     }
 
+    public void addThreshold(AbstractThreshold threshold) {
+        thresholds.add(threshold);
```

Comment:
you should set network element of the threshold here

Reply:
sure